### PR TITLE
Remove ert-storage as dependency

### PR DESCRIPTION
**Pull request by dafeda**
## Pre review checklist

- [ ] Read through the code changes carefully after finishing work
- [ ] Make sure tests pass locally (after every commit!)
- [ ] Prepare changes in small commits for more convenient review (optional)
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by frode-aarstad** [comment]
Should we also use the opportunity to rename `dark_storage` ?

File: src/ert/services/_storage_main.py [hunk 1]
```diff
@@ -121,20 +121,14 @@ def run_server(args: Optional[argparse.Namespace] = None, debug: bool = False) -
...
-        # that the import succeeds.
-        os.environ["ERT_STORAGE_DATABASE_URL"] = "sqlite://"
-        os.environ["ERT_STORAGE_RES_CONFIG"] = (
-            os.path.abspath(args.config) or find_ert_config()
-        )
-        config = uvicorn.Config("ert.dark_storage.app:app", **config_args)
+    # Dark Storage imports from ERT Storage, which connects to the database
+    # at startup. We set the database URL to an SQLite in-memory database so
+    # that the import succeeds.
+    os.environ["ERT_STORAGE_DATABASE_URL"] = "sqlite://"
```
Comment (frode-aarstad) [inline src/ert/services/_storage_main.py:127]: Think we can remove the DB references. There are also some references to `sqlite` around the code

**Review by pinkwah — COMMENTED** [review]
Please also put a link to the github repo and the commit from which you took this code, so that the git history is at least somewhat preserved.

**Comment by pinkwah** [comment]
> Should we also use the opportunity to rename `dark_storage` ?

I would like to at least add `_server` as postfix. I personally think `_storage_server` would be most descriptive of what this is supposed to be. Or `_onprem_storage_server`, to distinguish this from a possible future separate service that we would like to switch to.

**Comment by dafeda** [comment]
> Should we also use the opportunity to rename `dark_storage` ?

I'd rather we do that in a separate PR I think.

**Comment by dafeda** [comment]
> Please also put a link to the github repo and the commit from which you took this code, so that the git history is at least somewhat preserved.

I've squashed and added a link to the latest commit from ert-storage in the commit message body.

**Review by pinkwah — APPROVED** [review]

**Pull request merged**